src/t_string.c

```diff
@@ -265,107 +282,127 @@ int parseExtendedStringArgumentsOrReply(client *c, int *flags, int *unit, robj *
      */
     int cond_mut_excl = OBJ_SET_NX | OBJ_SET_XX | OBJ_SET_IFEQ | OBJ_SET_IFNE |
                         OBJ_SET_IFDEQ | OBJ_SET_IFDNE;
-
     for (; j < c->argc; j++) {
         char *opt = c->argv[j]->ptr;
         robj *next = (j == c->argc-1) ? NULL : c->argv[j+1];
 
         if ((opt[0] == 'n' || opt[0] == 'N') &&
             (opt[1] == 'x' || opt[1] == 'X') && opt[2] == '\0' &&
-            !(*flags & (cond_mut_excl & ~OBJ_SET_NX)) && (command_type == COMMAND_SET))
+            !(args->flags & OBJ_SET_XX) && (command_type == COMMAND_SET || command_type == COMMAND_MSETEX))
         {
-            *flags |= OBJ_SET_NX;
+            args->flags |= OBJ_SET_NX;
         } else if ((opt[0] == 'x' || opt[0] == 'X') &&
                    (opt[1] == 'x' || opt[1] == 'X') && opt[2] == '\0' &&
-                   !(*flags & (cond_mut_excl & ~OBJ_SET_XX)) && (command_type == COMMAND_SET))
+                   !(args->flags & OBJ_SET_NX) && (command_type == COMMAND_SET || command_type == COMMAND_MSETEX))
         {
-            *flags |= OBJ_SET_XX;
+            args->flags |= OBJ_SET_XX;
         } else if ((opt[0] == 'g' || opt[0] == 'G') &&
                    (opt[1] == 'e' || opt[1] == 'E') &&
                    (opt[2] == 't' || opt[2] == 'T') && opt[3] == '\0' &&
                    (command_type == COMMAND_SET))
         {
-            *flags |= OBJ_SET_GET;
-        } else if (!strcasecmp(opt, "KEEPTTL") && !(*flags & OBJ_PERSIST) &&
-            !(*flags & OBJ_EX) && !(*flags & OBJ_EXAT) &&
-            !(*flags & OBJ_PX) && !(*flags & OBJ_PXAT) && (command_type == COMMAND_SET))
+            args->flags |= OBJ_SET_GET;
+        } else if (!strcasecmp(opt, "KEYS") && command_type == COMMAND_MSETEX && j+1 < c->argc) {
+            /* Handle KEYS block skipping and populate KEYS parameters */
+            long kv_count_long;
+            if (getRangeLongFromObjectOrReply(c, c->argv[j+1], 1, INT_MAX,
+                &kv_count_long, "invalid numkeys value") != C_OK)
+            {
+                return C_ERR;
+            }
+            if (j + 2 + kv_count_long * 2 > c->argc) {
```

Comment:
i realized that there is a overflow risk here.
in 32bit, if kv_count_long == INT_MAX, kv_count_long * 2 will overflow.